src/main/java/com/atlassian/jira/cloud/jenkins/util/FreestyleBranchNameIssueKeyExtractor.java

```diff
@@ -17,22 +19,29 @@
 import hudson.scm.SCM;
 
 public class FreestyleBranchNameIssueKeyExtractor implements FreestyleIssueKeyExtractor {
-    private static final Logger logger = LoggerFactory.getLogger(FreestyleBranchNameIssueKeyExtractor.class);
 
     @Override
-    public Set<String> extractIssueKeys(final AbstractBuild<?, ?> freestyleBuild) {
+    public Set<String> extractIssueKeys(
+            final AbstractBuild<?, ?> freestyleBuild, final PipelineLogger pipelineLogger) {
         // The action is only injected for Freestyle jobs
         if (null != freestyleBuild.getProject() && null != freestyleBuild.getProject().getScm()) {
             SCM scm = freestyleBuild.getProject().getScm();
             GitSCM scm1 = (GitSCM) scm;
             List<BranchSpec> branches = scm1.getBranches();
-            Set<String> sets = IssueKeyStringExtractor.extractIssueKeys(branches.get(0).getName()).stream()
-                    .map(IssueKey::toString).collect(Collectors.toSet());
-
-            return sets;
+            Set<String> issueKeys =
+                    IssueKeyStringExtractor.extractIssueKeys(branches.get(0).getName())
+                            .stream()
+                            .map(IssueKey::toString)
+                            .collect(Collectors.toSet());
+
+            pipelineLogger.debug(
+                    String.format(
+                            "Extracted issue keys from branch name: %s",
+                            Arrays.toString(issueKeys.toArray())));
```

Comment:
I've seen this done in multiple places, doesn't `String.format` already do `toString` to all arguments and that `HashSet.toString` return `[entry1, entry2, etc]` automatically?

Reply:
When you call `toString()` on a list, it just prints out gibberish (the internal name of the object or something like that). Thus the workaround via `Arrays.toString()`...

Reply:
but it's not a list, it's a hashset?  I tried it with a small demo, unless issueKeys is not a set of strings?